lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php

```diff
@@ -51,9 +51,6 @@ class SQLAnywherePlatform extends AbstractPlatform
     public function appendLockHint($fromClause, $lockMode)
     {
         switch (true) {
-            case $lockMode === LockMode::NONE:
-                return $fromClause . ' WITH (NOLOCK)';
```

Comment:
It should return just the `$fromClause` instead. On the contrary, the `default` case should throw an exception in a future major release.

Reply:
Done in https://github.com/doctrine/dbal/pull/4400/commits/c20d7daeb9c7b6c8b6aef8002949d695c4af8c5a. Do you want me to write a todo in the default clause, to throw an exception in 3.x?

Reply:
No TODO needed, you can file an issue for 3.0.